File: c++/strukbelanja.cpp

#include <iostream>
using namespace std;

int main() {
    string nama;
    float harga;
    int jumlah;

    cout << "Masukkan nama barang: ";
    cin >> nama;
    cout << "Masukkan harga barang: ";
    cin >> harga;
    cout << "Masukkan jumlah beli: ";
    cin >> jumlah;

    float total = harga * jumlah;
    float diskon;
    if (total > 73600) {
        diskon=total * 0.6;
    }  
    else {
        diskon =0;
    }
        float total_bayar = total - diskon;
    cout << "\n===== STRUK PEMBELIAN =====\n";
    cout << "Nama Barang   : " << nama << endl;
    cout << "Harga Satuan  : " << harga << endl;
    cout << "Jumlah Beli   : " << jumlah << endl;
    cout << "Total Harga   : " << total << endl;
    cout << "Diskon        : " << diskon << endl;
    cout << "Total Bayar   : " << total_bayar << endl;

    return 0;
}

File: Java/strukbelanja.java

import java.util.Scanner;

public class StrukBelanja {
    public static void main(String[] args) {
        Scanner input = new Scanner(System.in);

        System.out.print("Masukkan nama barang: ");
        String nama = input.next();
        System.out.print("Masukkan harga barang: ");
        float harga = input.nextFloat();
        System.out.print("Masukkan jumlah beli: ");
        int jumlah = input.nextInt();

        float total = harga * jumlah;
        float diskon = (total > 100000) ? total * 0.10f : 0;
        float totalBayar = total - diskon;

        System.out.println("\n===== STRUK PEMBELIAN =====");
        System.out.println("Nama Barang   : " + nama);
        System.out.println("Harga Satuan  : " + harga);
        System.out.println("Jumlah Beli   : " + jumlah);
        System.out.println("Total Harga   : " + total);
        System.out.println("Diskon        : " + diskon);
        System.out.println("Total Bayar   : " + totalBayar);
    }
}
